src/r1cs_to_qap.rs

```diff
@@ -41,12 +41,23 @@ where
     return res;
 }
 
-pub(crate) struct R1CStoQAP;
+pub struct R1CStoQAP;
 
-impl R1CStoQAP {
+pub trait QAPCalculator {
```

Comment:
Maybe I would rename the trait to `R1CSToQAP`, and change the struct name to something more specific; maybe `LibsnarkReduction`, and the new circom one would be `CircomReduction`?